lib/glific/contacts/import.ex

```diff
@@ -37,18 +79,27 @@ defmodule Glific.Contacts.Import do
   The method takes in a csv file path and adds the contacts to the particular organization
   and group.
   """
-  @spec import_contacts(String.t(), integer, integer) :: tuple()
-  def import_contacts(file_path, organization_id, group_id) do
+  @spec import_contacts(integer, String.t(), []) :: tuple()
+  def import_contacts(organization_id, group_label, opts \\ []) do
+    if length(opts) > 1 do
+      raise "Please specify only one of keyword arguments: file_path, url or data"
+    end
+
+    contact_data_as_stream = fetch_contact_data_as_string(opts)
+
     with %{id: organization_id} <- Partners.organization(organization_id),
-         # this will raise an exception if group_id is not present
-         _group <- Groups.get_group!(group_id) do
-      result =
-        file_path
-        |> Path.expand()
-        |> File.stream!()
+         {_, group} <- Groups.get_or_create_group_by_label(group_label, organization_id) do
+      contact_data =
+        contact_data_as_stream
         |> CSV.decode(headers: true, strip_fields: true)
         |> Enum.map(fn {_, data} -> cleanup_contact_data(data, organization_id) end)
-        |> Enum.map(fn contact -> insert_or_update_contact_data(contact, group_id) end)
+
+      contact_data |> Enum.filter(fn contact -> contact.delete == "1" end) |> delete_contact_data
```

Comment:
any reason why we are not deleting one at a time, at some point for large files, we might move to streams, and doing it one at a time, will make things easier

Reply:
Yeah good call. I did not think it through. Gonna fix this :)